test/langtools/tools/javap/TestClassNameWarning.java

```diff
@@ -177,7 +177,7 @@ public void testNoNameClass(Path base) throws Exception {
         ClassFile cf2 = new ClassFile(
                 cf.magic, cf.minor_version, cf.major_version, cf.constant_pool,
                 cf.access_flags,
-                0, // this_class,
+                cf.this_class,
```

Comment:
Should we just remove this no-name-class case since it's already an invalid class file?

Reply:
I've fixed this test so it reflects actual javap behaviour.